bin/hdfcoinc/pycbc_multiifo_add_statmap

```diff
@@ -0,0 +1,214 @@
+
+# output to file
+f.attrs['foreground_time'] = abs(foreground_segs)
+
+logging.info('Copying foreground datasets')
+for k in files[0]['foreground']:
+    if not k.startswith('fap') and k not in all_ifos:
+        pycbc.io.combine_and_copy(f, files, 'foreground/' + k)
+
+logging.info('Collating triggers into single structure')
+
+all_trig_times = {}
+all_trig_ids = {}
+for ifo in all_ifos:
+    all_trig_times[ifo] = np.array([], dtype=np.uint32)
+    all_trig_ids[ifo] = np.array([], dtype=np.uint32)
+
+# For each file, append the trigger time and id data for each ifo
+# If an ifo does not participate in any given coinc then fill with -1 values
+for f_in in files:
+    for ifo in all_ifos:
+        if ifo in f_in['foreground']:
+            all_trig_times[ifo] = np.concatenate([all_trig_times[ifo], \
+                                    f_in['foreground/{}/time'.format(ifo)][:]])
+            all_trig_ids[ifo] = np.concatenate([all_trig_ids[ifo],
+                              f_in['foreground/{}/trigger_id'.format(ifo)][:]])
+        else:
+            all_trig_times[ifo] = np.concatenate([all_trig_times[ifo],
+                                 -1*np.ones_like(f_in['foreground/fap'][:],
+                                                    dtype=np.uint32)])
+            all_trig_ids[ifo] = np.concatenate([all_trig_ids[ifo],
+                                 -1*np.ones_like(f_in['foreground/fap'][:],
+                                                    dtype=np.uint32)])
+
+for ifo in all_ifos:
+    f['foreground/{}/time'.format(ifo)] = all_trig_times[ifo]
+    f['foreground/{}/trigger_id'.format(ifo)] = all_trig_ids[ifo]
+
+n_triggers = f['foreground/ifar'].size
+logging.info('{} triggers'.format(n_triggers))
+
+# all_times is a tuple of trigger time arrays
+all_times = (f['foreground/%s/time' % ifo][:] for ifo in all_ifos)
+
+# Currently only clustering zerolag, i.e. foreground, so set all timeslide_ids
+# to zero
+cidx = pycbc.events.cluster_coincs_multiifo(f['foreground/stat'][:], all_times,
+                                            np.zeros(n_triggers), 0,
+                                            args.cluster_window, argmax=np.argmax)
+
+
+def filter_dataset(h5file, name, idx):
+    # Dataset needs to be deleted and remade as it is a different size
+    filtered_dset = h5file[name][:][idx]
+    del h5file[name]
+    h5file[name] = filtered_dset
+
+# Downsample the foreground columns to only the loudest ifar between the
+# multiple files
+for key in f['foreground'].keys():
+    if key not in all_ifos:
+        filter_dataset(f, 'foreground/%s' % key, cidx)
+    else:  # key is an ifo
+        for k in f['foreground/%s' % key].keys():
+            filter_dataset(f, 'foreground/{}/{}'.format(key, k), cidx)
+
+n_triggers = f['foreground/ifar'].size
+
+times_tuple = (f['foreground/{}/time'.format(ifo)] for ifo in all_ifos)
+test_times = np.array([pycbc.events.mean_if_greater_than_zero(tc)[0]
```

Comment:
perhaps put a comment before this block eg 'Calculating event times to determine which types of coinc are active'